tools/fw-update/rs-fw-update.cpp

```diff
@@ -119,7 +121,57 @@ void list_devices(rs2::context ctx)
     }
 }
 
-int main(int argc, char** argv) try
+int write_fw_to_mipi_device( const std::vector< uint8_t > & fw_image)
+{
+    // Write firmware to appropriate file descritptor
+    std::cout << std::endl << "Update can take up to 2 minutes" << std::endl;
+    std::ofstream fw_path_in_device( "/dev/d4xx-dfu504", std::ios::binary );
+    auto file_deleter = std::unique_ptr< std::ofstream, void ( * )( std::ofstream * ) >( &fw_path_in_device,
+                                                                                         []( std::ofstream * file )
+                                                                                         { file->close(); } );
+    if( fw_path_in_device )
+    {
+        bool done = false;
+        std::thread show_progress_thread(
+            [&done]()
+            {
+                for( int i = 0; i < 101 && ! done; ++i ) // Show percentage [0-100]
+                {
+                    printf( "%d%%\r", i );
+                    std::cout.flush();
+                    std::this_thread::sleep_for( std::chrono::seconds( 1 ) );
+                }
+            } );
+        fw_path_in_device.write( reinterpret_cast< const char * >( fw_image.data() ), fw_image.size() );
```

Comment:
Please verify if we can notify when fail on write

Reply:
Done